test/stat.jl

```diff
@@ -0,0 +1,84 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+using Base.Filesystem: StatStruct, samefile
+
+show(DevNull, StatStruct())
+mktempdir() do d
+    mktemp(d) do f, io
+        @test typeof(stat(RawFD(fd(io)))) <: StatStruct
+        @test typeof(stat(fd(io))) <: StatStruct
+        @test typeof(stat(splitdir(f)...)) <:StatStruct
+        @test typeof(lstat(splitdir(f)...)) <: StatStruct
+        @test ispath(d)
+        @test !ispath(joinpath(d, randstring(17)))
+        if is_unix()
+            ff = joinpath(d, string(randstring(17), "_fifo"))
+            run(`mkfifo $ff`)
+            @test isfifo(ff)
+            @test !isfifo(f)
+            @test ischardev("/dev/stderr")
+            @test !ischardev(d)
+        end
+        @test isdir(d)
+        @test !isdir(f)
+        # Portable suggestions welome.
+        if is_linux()
+            @test isblockdev("/dev/ram0")
+            @test !isblockdev("/")
+        end
+        @test isfile(f)
+        @test !isfile(d)
+        if is_unix()
+            fl = string(f, "_symlink")
+            symlink(f, fl)
+            @test !islink(f)
+            @test islink(fl)
+            @test samefile(f, fl)
+            mktemp(d) do f′, _
+                @test !samefile(f′, f)
+                @test !samefile(f′, fl)
+            end
+        end
+        @test samefile(f, f)
+        @test samefile(d, d)
+        @test !samefile(f, d)
+        if is_unix()
```

Comment:
they should work, they just require a specific formatting to the file name (they can't be put in an arbitrary directory)